Source/Common/EnvironmentLoader.cpp

```diff
@@ -0,0 +1,108 @@
+#include <functional>
+#include <string_view>
+#include <unordered_map>
+
+#include "Common/Config.h"
+
+
+
+namespace FEX::EnvLoader {
+
+  using string = std::string;
+  using string_view = std::string_view;
+
+  void Load(char *const envp[])
+  {
+    std::unordered_map<string_view,string_view> EnvMap;
+
+    for(const char *const *pvar=envp; pvar && *pvar; pvar++) {
+      string_view Var(*pvar);
+      size_t pos = Var.rfind('=');
+      if (string::npos==pos)
+        continue;
+
+      string_view Ident = Var.substr(0,pos);
+      string_view Value = Var.substr(pos+1);
+      EnvMap[Ident]=Value;
+    }
+
+    std::function GetVar = [=](const string_view id)  -> const string_view {
+      if (EnvMap.find(id) != EnvMap.end()) 
+        return EnvMap.at(id);
+
+      // If envp[] was empty, search using std::getenv()
+      const char* vs = std::getenv(id.data());
+      string_view sv(vs?vs:"");
+      return sv;
+    };
+
+    string_view Value;
+    if ((Value = GetVar("FEX_CORE")).size()) {
+      // Accept Numeric or name //
+      if(isdigit(Value[0])) Config::Add("Core", Value);
+      else {
+        uint32_t CoreVal = 0;
+             if (Value==string_view("irint")) CoreVal=0; // default
+        else if (Value==string_view("irjit")) CoreVal=1;
+        else if (Value==string_view("llvm"))  CoreVal=2;
+        else if (Value==string_view("host"))  CoreVal=3;
+        else if (Value==string_view("vm"))    CoreVal=4;
+        else { assert(0); }
+        Config::Add("Core", std::to_string(CoreVal));
+      }
+    }
+
+    if ((Value = GetVar("FEX_BREAK")).size()) {
+      if(isdigit(Value[0]))
```

Comment:
I know we don't have a coding guideline yet, but would be nice if this is either on the same line as the `if` or if you want to retain it on the next line. Wrap it in curly braces